### PR TITLE
feat: Circular Progress in Landscape

### PR DESCRIPTION
**Pull request by shadow578**
This PR adds support for using a circular progress bar in landscape mode.
To facilitate this, the progress bar layout (`progress_bar.xml`) has been changed from a `LinearLayout` to a `ConstraintLayout` with nested `FrameLayout`. 

In addition, the progress bar style settings have been split into individual portrait and landscape mode settings.
This allows users to choose between the progress bar styles `None`, `Linear` and `Circular` for each screen orientation.

As selecting the `None` style for landscape mode has the same effect as the `Disable in Landscape` option, it has been removed.

| buck_landscape |
|-|
| showcase of circular progress bar in landscape mode |

new-settings | style-popup
-|-
individual settings for portrait and landscape orientation | progress bar style options

> [!NOTE]
> i tested this on a Google Pixel 6a, but other devices should work too.

### CODE REVIEW

**Comment by GustavoASantos** [comment]
This is awesome, thanks for contributing! I like the icon you picked for the landscape progress bar style 😃

File: app/src/main/java/com/gustavoas/noti/services/AccessibilityService.kt [hunk 1]
```diff
@@ -155,26 +161,145 @@ class AccessibilityService : AccessibilityService() {
...
+        val progressParams = circularProgressBar.layoutParams as FrameLayout.LayoutParams
+
+        // reset container constraints
+        containerParams.topToTop = -1
+        containerParams.bottomToBottom = -1
+        containerParams.leftToLeft = -1
+        containerParams.rightToRight = -1
+
+        // add navbar height to container margins
+        containerParams.setMargins(0, 0, 0, -navigationBarHeight)
```
Comment (GustavoASantos) [inline app/src/main/java/com/gustavoas/noti/services/AccessibilityService.kt:175]: If it's only using the display over other apps permission and not the accessibility service it won't be able to get the navigationBarHeight here before the view is attached

```suggestion
        overlayView.doOnAttach {
                containerParams.setMargins(0, 0, 0, -navigationBarHeight)
        }
```

File: app/src/main/java/com/gustavoas/noti/services/AccessibilityService.kt [hunk 2]
```diff
@@ -215,10 +340,27 @@ class AccessibilityService : AccessibilityService() {
     }
 
     private fun isLocked(): Boolean {
-        val keyguardManager = getSystemService(KEYGUARD_SERVICE) as KeyguardManager
         return keyguardManager.isKeyguardLocked
     }
 
+    private val statusBarHeight: Int
+        get() {
+            return ViewCompat.getRootWindowInsets(overlayView)?.getInsets(WindowInsetsCompat.Type.statusBars())?.top ?: 0
```
Comment (GustavoASantos) [inline app/src/main/java/com/gustavoas/noti/services/AccessibilityService.kt:348]: The statusBarHeight is always returning 0 when using the accessibility service

**Review by GustavoASantos — CHANGES_REQUESTED** [review]
Great job! I noticed a couple of details that need changing but other than that it's working perfectly. If you don't yet have Noti on Google Play send me an email to gustavoasgas1@gmail.com

File: app/src/main/java/com/gustavoas/noti/services/AccessibilityService.kt (hunk 1 above)
Comment (shadow578) [reply app/src/main/java/com/gustavoas/noti/services/AccessibilityService.kt:175]: > If it's only using the display over other apps permission and not the accessibility service it won't be able to get the navigationBarHeight here before the view is attached

that is correct. I'll add your suggestion


> This also seems to not be needed when using 3-button navigation

i was under the impression that the navigation bar with 3-button navigation will remain on the bottom edge of the screen, but at least on my pixel 6a this is not the case (the navigation bar remains on the side of the screen, where the charging port is). 

I'll add a check for this

File: app/src/main/java/com/gustavoas/noti/services/AccessibilityService.kt (hunk 2 above)
Comment (shadow578) [reply app/src/main/java/com/gustavoas/noti/services/AccessibilityService.kt]: i've reverted it to the way it worked before the PR

File: app/src/main/java/com/gustavoas/noti/services/AccessibilityService.kt (hunk 1 above)
Comment (shadow578) [reply app/src/main/java/com/gustavoas/noti/services/AccessibilityService.kt]: i've reworked the `navigationbarHeight` function and it should now handle 3-button navigation

**Comment by GustavoASantos** [comment]
Seems to be working perfectly now. Can I merge it?

**Comment by shadow578** [comment]
> Seems to be working perfectly now. Can I merge it?

absolutely. I've also done some more testing and it seems to work for me too.

> If you don't yet have Noti on Google Play send me an email to gustavoasgas1@gmail.com

i'll be sending you a mail shortly. I assume you need the mail address of my google account?

**Comment by GustavoASantos** [comment]
> I assume you need the mail address of my google account?

No need, I'll send you a code which you can redeem in Google Play

**Comment by shadow578** [comment]
> > I assume you need the mail address of my google account?
> 
> No need, I'll send you a code which you can redeem in Google Play

i've sent the email just now

**Pull request merged**